andela/bp-esa-backend
# #160436611 Slack channel services

#### What does this PR do?
Enables users to create new slack channel, add developer to channel, and remove a developer from slack channel

#### Description of Task to be completed?
- Generate formatted channel names based on partner names
- Create private slack channels for new engagements
- Add placed developer to engagement slack channel
- Remove rolled off developer from partner slack channel


#### How should this be manually tested?
Assuming you are an administrator of the workspace:

- Create an app. It should have workspace privileges in order to create channels(groups:write)
- Generate a token for the app and add it to the .env file as the SLACK_TOKEN
- Call the `createPartnerChannels` function in `server/modules/slackIntegration.js`, while providing the partner's details, to create a **private** channel for a specific partner engagement
- Call the addToChannel function in server/modules/slackIntegration.js, while providing the user's email and channel ID, to add user to channel.
- Call the `removeFromChannel` function in server/modules/slackIntegration.js, while providing the user's email and channel ID, to remove a user from channel.

#### Background context
- Placed developers should be removed from the `available-devs` channel when placed.

#### What are the relevant pivotal tracker stories?
[160436728](https://www.pivotaltracker.com/story/show/160436728), [160436783](https://www.pivotaltracker.com/story/show/160436783), [160436611](https://www.pivotaltracker.com/story/show/160436611)
